### PR TITLE
Add tempest job for glance-operator

### PR DESCRIPTION
**Pull request by bshewale**

### CODE REVIEW

File: ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-master.yaml [hunk 1]
```diff
@@ -6,7 +6,11 @@ base_images:
   golangci-lint:
     name: golangci-lint
     namespace: ci
-    tag: v1.48.0
+    tag: v1.46.0
```
Comment (marios) [inline ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-master.yaml:9]: as discussed this is a rebase nit needs update

File: ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-master.yaml [hunk 2]
```diff
@@ -74,6 +78,19 @@ tests:
...
+    architecture: amd64
+    cloud: vexxhost
+    owner: openstack-k8s-operators
+    product: ocp
+    timeout: 1h20m0s
+    version: "4.10"
+  optional: true
+  steps:
+    env:
+      TEMPEST_REGEX: tempest.api.image.v2
```
Comment (marios) [inline ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-master.yaml:92]: and this one needs to be removed before merge as discussed with @arxcruz

**Review by marios — CHANGES_REQUESTED** [review]
needs update

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-master-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-master-glance-operator-build-deploy-tempest

**Comment by fmount** [comment]
I think it's worth investigating the failure [1] and see if we require some
tuning in the regex or we need any test blacklisted (e.g. we have a known
issue and we want to whitelist the test once fixed in tempest).
@konan-abhi can you help/take a look at [1] ?


[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37104/rehearse-37104-pull-ci-openstack-k8s-operators-glance-operator-master-glance-operator-build-deploy-tempest/1653380439496200192/artifacts/glance-operator-build-deploy-tempest/openstack-k8s-operators-tempest/artifacts/stestr_results.html

**Comment by konan-abhi** [comment]
> I think it's worth investigating the failure [1] and see if we require some tuning in the regex or we need any test blacklisted (e.g. we have a known issue and we want to whitelist the test once fixed in tempest). @konan-abhi can you help/take a look at [1] ? @bshewale can we rebase the PR?
> 
> [1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37104/rehearse-37104-pull-ci-openstack-k8s-operators-glance-operator-master-glance-operator-build-deploy-tempest/1653380439496200192/artifacts/glance-operator-build-deploy-tempest/openstack-k8s-operators-tempest/artifacts/stestr_results.html

In devstack environment during installation/setup we run command [1] to load the available namespaces into database which reads those from [2]. 

[1] https://github.com/openstack/devstack/blob/master/lib/glance#L492 
[2] https://github.com/openstack/glance/tree/master/etc/metadefs

I think we miss this step here in operator setup and that is why we are getting empty list in response and further failure for the same.

**Comment by fmount** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-master-glance-operator-build-deploy-tempest

**Comment by fmount** [comment]
@bshewale do we want to rebase this patch and retrigger tempest?
The glance-operator PR https://github.com/openstack-k8s-operators/glance-operator/pull/232 merged, so I think now it would be interesting re-kicking tempest and observe any potential different behavior.

File: ci-operator/jobs/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-main-presubmits.yaml [hunk 3]
```diff
@@ -124,6 +124,67 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )glance-operator-build-deploy-kuttl,?($|\s.*)
+- agent: kubernetes
```
Comment (fmount) [inline ci-operator/jobs/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-main-presubmits.yaml:127]: @bshewale we should probably have an issue on this file [1]

[1] https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/37104/pull-ci-openshift-release-master-config/1664246351929544704

**Review by fmount — CHANGES_REQUESTED** [review]

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-master-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

File: ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-main.yaml [hunk 4]
```diff
@@ -3,6 +3,14 @@ base_images:
     name: "4.10"
     namespace: ocp
     tag: cli
+  golangci-lint:
```
Comment (Sandeepyadav93) [inline ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-main.yaml:6]: Where we will use golangci-lint container

**Review by Sandeepyadav93 — CHANGES_REQUESTED** [review]

File: ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-main.yaml (hunk 4 above)
Comment (frenzyfriday) [reply ci-operator/config/openstack-k8s-operators/glance-operator/openstack-k8s-operators-glance-operator-main.yaml:6]: removing this

**Comment by fmount** [comment]
mm I don't see the job triggered after the last update

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by Sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by fmount** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Comment by frenzyfriday** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-tempest

**Review by fmount — APPROVED** [review]
+1, tests are green, I guess we can merge this change.

**Pull request merged**